src/components/common/ProductBox/ProductBox.js

```diff
@@ -16,6 +16,15 @@ const addFurnitureToFavourite = (id, isFavourite, event, addFavourite) => {
   addFavourite({ id: id, isFavourite: !isFavourite });
 };
 
+const addFurnitureToCompare = (id, compare, event, addCompare, inCompare) => {
+  event.preventDefault();
+  if (inCompare < 4) {
+    addCompare({ id: id, compare: !compare });
+  } else {
+    addCompare({ id: id, compare: false });
```

Comment:
Po co w ogóle wywoływać `addCompare` skoro i tak nie zostaną dodane?

Reply:
Tutaj użyłem addCompare bo zmienia stan i w przypadku kliknięcia na wczesniej wybrany produkt zmieni stan i produkt wyskoczy z belki. Zrobiłem tak:

const addFurnitureToCompare = (id, compare, event, addCompare, inCompare) => {
  event.preventDefault();
  if (inCompare < 4) {
    addCompare({ id: id, compare: !compare });
  } else {
    return {compare : false} ;
  }
};
ale od kliknięcie ikony porównania nie zmienia stanu. Nie wiem jak to inaczej zrobić. Generalnie nazwa jest myląca bo wskazuje ze funkcja ma tylko dodać do compare ale w rzeczywistości ta funkcja dodaje i usuwa wczesniej dodane elementy.

Reply:
tak tez nie działa:

if (inCompare < 4) {
    return{ id: id, compare: !compare };
  } else {
    return{ id: id, compare: false };
  }

Reply:
a w jaki sposób `inCompare < 4` jest połączone z tym czy dany produkt był kliknięty czy nie?? 
`inCompare` to ilość produktów dodana do porównania a nie info czy ten konkretny jest porównywany.

Tu raczej jest Ci potrzebny if czy `compare === true` jeśli tak to `removeCOmpare` jeśli nie to `addCompare`

Reply:
porpawiłem, są dwie osbne funkcje addCompare i removeCompare, inCompare zamieniłem na flage canAddCompare. Sporo to uprościło logike no i jest czytelniej